File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.cs
public class Solution {
    public IList<int> FindDisappearedNumbers(int[] nums) {
        HashSet<int> hs = nums.ToHashSet();
        List<int> list = new(nums.Length - hs.Count);
        for(int i = 1, j = 0; i <= nums.Length; i++){
            if(!hs.Contains(i)){
                list.Add(i);
            }
        }
        return list;
        
    }
}

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        vector<int> ans;
        int len = nums.size();
        int cnt = 1;
        sort(nums.begin(), nums.end());
        for(int i = 0; i < nums.size(); i++){
            if(cnt == nums[i]){
                cnt++;
            }else if(cnt < nums[i]){
                cout << "asd : " << nums[i] << endl;
                while(cnt < nums[i]){
                    ans.push_back(cnt);
                    cnt++;
                    if(cnt == nums[i]) cnt++;
                }
            }
        }
        while(cnt <= len){
            ans.push_back(cnt++);
        }
        return ans;
    }
};